src/ncp/ncp_base.hpp

```diff
@@ -588,12 +602,49 @@ class NcpBase
     uint8_t mPreferredRouteId;
 #endif
 
+    uint8_t mCurCommandIID;
+
 #if OPENTHREAD_RADIO || OPENTHREAD_CONFIG_LINK_RAW_ENABLE
+    uint8_t mCurTransmitIID;
     uint8_t mCurTransmitTID;
     int8_t  mCurScanChannel;
     bool    mSrcMatchEnabled;
 #endif // OPENTHREAD_RADIO || OPENTHREAD_CONFIG_LINK_RAW_ENABLE
 
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+#if OPENTHREAD_RADIO || OPENTHREAD_CONFIG_LINK_RAW_ENABLE
+#define kPendingCommandQueueSize SPINEL_HEADER_IID_MAX
+
+    enum PendingCommandType
+    {
+        kPendingCommandTypeTransmit,
+        kPendingCommandTypeEnergyScan,
+    };
```

Comment:
A general question on pending command model?

I understand that we enqueue "Tx" and "Energy Scan" request from different IIDs and handle them sequentially to avoid conflicting commands to radio.

What about other conflicting commands from different IIDS (different processes on host trying to control RCP/radio)? An example:
   - Suppose IID 1  and 2 both request a frame TX, we have IID 1 TX and reposne TX done back to it and then start TX for IID2
   - What if after TX done, IID 1 tries to put radio to sleep or receive mode (on different channel)? How will it be handled?
       - IID 1 can issue `Set(SPINEL_PROP_MAC_RAW_STREAM_ENABLED, false)` to put radio to sleep which we will process when we receive the spinel command on RCP and call `otLinkRawSleep()` which will try to put radio to sleep.
   - But while radio is in middle of TX for IID2, we cannot really put it to sleep or rx mode  on different channel?

Reply:
It is a valid scenario, but currently, we don't support sleep or different channel operations (other than scanning) with multipan feature. But we will re-visit/consider this for future enhancements or when we extend this for other features that need those functionalities.